### PR TITLE
chore(cloud-sources): add telemetry for cloud sources

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

Add telemetry for the number of cloud source integrations as well as the different types.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

N/A.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* **#10053** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/10053?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/10053?utm_source=stack-comment).

File: central/cloudsources/datastore/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Integration),
+		),
+	)
+	props := make(map[string]any)
+
+	cloudSources, err := Singleton().ListCloudSources(ctx, search.EmptyQuery())
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get image integrations")
```
Comment (stehessel) [inline central/cloudsources/datastore/telemetry.go:37]:
```suggestion
		return nil, errors.Wrap(err, "failed to list cloud sources")
```

**Review by stehessel — APPROVED** [review]

**Comment by 0x656b694d** [comment]
Do you mind running a local test targeting the development telemetry project?

File: central/cloudsources/datastore/telemetry.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+var Gather phonehome.GatherFunc = func(ctx context.Context) (map[string]any, error) {
+	ctx = sac.WithGlobalAccessScopeChecker(ctx,
+		sac.AllowFixedScopes(
+			sac.AccessModeScopeKeys(storage.Access_READ_ACCESS),
+			sac.ResourceScopeKeys(resources.Integration),
+		),
+	)
+	props := make(map[string]any)
+
+	cloudSources, err := Singleton().ListCloudSources(ctx, search.EmptyQuery())
```
Comment (janisz) [inline central/cloudsources/datastore/telemetry.go:35]: I we get this as a param then we can unit test this function

**Comment by dhaus67** [comment]
@0x656b694d can confirm with a local test on the dev telemetry key that we receive the expected properties: 
Screenshot 2024-02-22 at 17 13 50

**Pull request merged**